src/torchcodec/samplers/_implem.py

```diff
@@ -339,14 +339,25 @@ def clips_at_regular_indices(
 def _validate_params_time_based(
     *,
     decoder,
+    num_clips,
     seconds_between_clip_starts,
     seconds_between_frames,
 ):
-    if seconds_between_clip_starts <= 0:
+
+    if (num_clips is None and seconds_between_clip_starts is None) or (
+        num_clips is not None and seconds_between_clip_starts is not None
+    ):
+        # This is internal only and should never happen
+        raise ValueError("Bad, bad programmer!")
```

Comment:
Can you add more details to this message, like the actual values of num_clips and seconds_between...

Reply:
I will change the error message to something more conventional, but to repeat the comment: this is non-user facing code, it's purely internal and should never *ever* be triggered